### PR TITLE
[WIP] Add ORHorizontalStackView

### PR DESCRIPTION
**Pull request by garnett**
- [x] Add ```direction``` property
- [x] Rename public APIs
- [x] Update inline documentation to reflect both vertical/horizontal layout
- [x] ```ORStackScrollView``` won't work properly if we change ```direction``` of it's internal stackView (because of obsolete vertical constraints)

### CODE REVIEW

**Comment by garnett** [comment]
@orta I'm little bit stuck: 
```ORTagBasedAutoStackView``` looks like it fully violates Liscov substitution principle. IMO it should not be sibiling of ```ORStackView```` + it also should have horizontal counterpart. It leads us to on more class if we plan to use composition.
Situation with ```ORStackScrollView``` is also tricky - it should know how to behave depending of layout direction (grow bottom or right).

I am thinking again about 'all in one solution', by introducing new terminology to existing APIs:
for example we can use 'beforeMargin' to cover both 'topMargin' and 'leadingMargin' + use ```insertSubviewAfterSubview``` and ```insertSubviewBeforeSubview``` to unify vertical and horizontal interfaces. What do you think?

**Comment by orta** [comment]
> I am thinking again about 'all in one solution', by introducing new terminology to existing APIs:
for example we can use beforeMargin to cover both topMargin and leadingMargin (for horizontal layout) + use insertSubviewAfterSubview and insertSubviewBeforeSubview to unify vertical and horizontal interfaces. What do you think?

This feels reasonable to me, 3.0.0 has already declared API bankruptcy. Designing it as though it does not have to deal with backwards compatibility in mind is totally :+1:

**Comment by garnett** [comment]
- [x] Add ```direction```
- [x] Rename public APIs
- [] Update inline documentation to reflect both vertical/horizontal layout
- [] ```ORStackScrollView``` won't work properly if we change ```direction of it's internal stackView (because of obsolete vertical constraints)

**Comment by orta** [comment]
/// Adds a view to the heirarchy, if it's the first it is
/// attached to the top, otherwise to the view above.

> Adds a view to the hierarchy, depending on the orientation it will place it at the the top of the stack either vertically or horizontally. After this  views will be attached incrementally.

**Comment by orta** [comment]
> ORStackScrollView won't work properly if we change direction of it's internal stackView (because of obsolete vertical constraints)

I think the view hierarchy get completely dropped and re-recreated on setupConstraints, meaning this won't be a problem?

**Comment by garnett** [comment]
@orta please review, I've updated scrollview and documentation.

File: Classes/ios/ORStackView.h [hunk 1]
```diff
@@ -9,37 +9,43 @@
...
-- (void)addSubview:(UIView *)view withTopMargin:(CGFloat)margin;
+/// Adds a view to the hierarchy, depending on the orientation it will place it
+/// at the the top of the stack either vertically or horizontally.
+/// After this views will be attached incrementally.
+- (void)addSubview:(UIView *)view withStartMargin:(CGFloat)margin;
 
 /// Adds a view to the heirarchy like addSubview:withTopMargin:
 /// will also center and apply side margins as insets from edge
-- (void)addSubview:(UIView *)view withTopMargin:(CGFloat)topMargin sideMargin:(CGFloat)sideMargin;
+- (void)addSubview:(UIView *)view withStartMargin:(CGFloat)topMargin sideMargin:(CGFloat)sideMargin;
```
Comment (orta) [inline Classes/ios/ORStackView.h]: I like this, another option is `initialMargin` :+1:

Comment (orta) [inline Classes/ios/ORStackView.h]: OK, I think `precedingMargin` describes it well actually

File: ORStackViewExample/ORFirstViewController.m [hunk 2]
```diff
@@ -30,39 +30,51 @@ - (void)loadView
...
     [self.view addSubview:self.stackView];
     if ([self respondsToSelector:@selector(topLayoutGuide)]) {
         [self.view addConstraint:[NSLayoutConstraint constraintWithItem:self.stackView attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:self.topLayoutGuide attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0]];
+    } else {
+        [self.view addConstraint:[NSLayoutConstraint constraintWithItem:self.stackView attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:self.view attribute:NSLayoutAttributeTop multiplier:1.0 constant:0]];
     }
-    [self.view addConstraint:[NSLayoutConstraint constraintWithItem:self.stackView attribute:NSLayoutAttributeWidth relatedBy:NSLayoutRelationEqual toItem:self.view attribute:NSLayoutAttributeWidth multiplier:1.0 constant:0]];
-    [self.view addConstraint:[NSLayoutConstraint constraintWithItem:self.stackView attribute:NSLayoutAttributeCenterX relatedBy:NSLayoutRelationEqual toItem:self.view attribute:NSLayoutAttributeCenterX multiplier:1.0 constant:0]];
+    
+    if ([self respondsToSelector:@selector(bottomLayoutGuide)]) {
```
Comment (orta) [inline ORStackViewExample/ORFirstViewController.m]: :+1:

**Comment by orta** [comment]
This feels good to me @garnett :tada: 

If I can be as rude as to ask to change to `precedingMargin` then I'm totally 100% :D

**Comment by garnett** [comment]
@orta sure, I find ```precedingMargin``` to be nicer synonym here

**Comment by orta** [comment]
Ace, this looks good to me. Merrrrrrrgin'

**Pull request merged**